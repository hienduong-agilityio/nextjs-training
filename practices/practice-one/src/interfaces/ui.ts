export interface IInputGroupProps {
  placeholder?: string;
  value?: string;
  buttonText?: string;
  inputName?: string;
  isDisabled?: boolean;
  isLoading?: boolean;
  startIcon?: React.ReactNode;
  customClass?: {
    container?: string;
    inputContainer?: string;
    input?: string;
    button?: string;
  };
  onSubmit?: (event: React.FormEvent<HTMLFormElement>) => void;
}
