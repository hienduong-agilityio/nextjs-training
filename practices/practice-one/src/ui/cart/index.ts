export * from './CartTable';
export * from './OrderSummary';
export * from './VoucherInput';
export * from './EmptyCart';

export { default as CartItemRow } from './CartItemRow';
