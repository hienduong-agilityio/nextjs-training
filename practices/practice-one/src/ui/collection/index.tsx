export * from './LoadingCategoryGroup';
export * from './LoadingFilterSortBar';
