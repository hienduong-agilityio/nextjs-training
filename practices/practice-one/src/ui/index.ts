export * from './products/index';
export * from './collection/index';
