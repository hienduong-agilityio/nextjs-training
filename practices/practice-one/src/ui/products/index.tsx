export * from './CartAndFavoriteActions';
export * from './ProductImages';
export * from './ProductInfo';
export { default as QuantityControl } from './QuantityControl';
export * from './SocialShare';
export * from './RelatedProducts';
export * from './ProductDetailTabs';
export * from './LoadingRelatedProducts';
export * from './LoadingProductCard';
export * from './LoadingProductTabs';
export * from './ProductReview';
