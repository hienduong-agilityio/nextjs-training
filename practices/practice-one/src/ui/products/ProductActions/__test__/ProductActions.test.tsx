import { render, screen, fireEvent } from '@testing-library/react';

// UI
import { ProductActions } from '@/ui';

describe('ProductActions Component', () => {
  const productId = 'test-product-id';
  let quantityInput: HTMLElement;
  let incrementButton: HTMLElement;
  let decrementButton: HTMLElement;

  beforeEach(() => {
    render(<ProductActions productId={productId} />);

    quantityInput = screen.getByLabelText('Quantity input');
    incrementButton = screen.getByLabelText('increment');
    decrementButton = screen.getByLabelText('decrement');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<ProductActions productId={productId} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with initial state', () => {
    expect(screen.getByLabelText('Quantity Control')).toBeInTheDocument();
    expect(
      screen.getByLabelText('Cart and Favorite Actions'),
    ).toBeInTheDocument();
  });

  it('updates quantity when handleQuantityChange is called via QuantityControl', () => {
    fireEvent.click(incrementButton);
    expect(quantityInput).toHaveValue(2);

    fireEvent.click(decrementButton);
    expect(quantityInput).toHaveValue(1);
  });
});
