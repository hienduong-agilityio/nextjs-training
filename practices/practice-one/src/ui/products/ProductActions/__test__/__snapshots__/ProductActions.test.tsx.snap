// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductActions Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col items-center justify-between gap-4 mt-4 md:flex-row py-6 border-y-2 border-secondary-100"
  >
    <div
      aria-label="Quantity Control"
      class="rounded-md gap-4 bg-secondary-50"
    >
      <div
        class="flex"
      >
        <button
          aria-label="decrement"
          class="px-5 py-3 text-primary-100 hover:text-gray-800 disabled:cursor-not-allowed disabled:text-gray-800"
          disabled=""
          type="button"
        >
          -
        </button>
        <input
          aria-label="Quantity input"
          class="w-12 text-center border-0 bg-gray-50 focus:ring-0"
          type="number"
          value="1"
        />
        <button
          aria-label="increment"
          class="px-5 py-3 text-primary-100 hover:text-gray-800 disabled:cursor-not-allowed disabled:text-gray-800"
          type="button"
        >
          +
        </button>
      </div>
    </div>
    <div
      aria-label="Cart and Favorite Actions"
      class="flex gap-3 xl:gap-5"
    >
      <button
        aria-label="Add to Cart"
        class="flex items-center justify-center gap-2 rounded p-2 disabled:cursor-not-allowed disabled:opacity-50 bg-opacity-100 group shadow-none bg-primary-50 text-primary-100 disabled:hover:text-primary-100 disabled:hover:bg-primary-50 hover:bg-primary-100 hover:text-white transition-all duration-200 whitespace-nowrap"
        type="button"
      >
        <span>
          <svg
            class="text-primary-100 group-hover:text-white transition-colors duration-200"
            fill="none"
            height="20"
            viewBox="0 0 20 20"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M15.22 12.705c.526 0 1.036-.18 1.442-.507.407-.326.684-.78.786-1.285L18.59 5.34a2.176 2.176 0 0 0-.023-.976 2.208 2.208 0 0 0-.449-.872 2.276 2.276 0 0 0-.787-.602 2.33 2.33 0 0 0-.975-.216H4.988V1.559c0-.295-.12-.578-.333-.786a1.156 1.156 0 0 0-.804-.328H1.582c-.297.009-.578.13-.785.338a1.101 1.101 0 0 0-.323.777c0 .29.116.568.323.776.207.208.488.33.785.338h1.142v11.145a2.323 2.323 0 0 0-1.271.37c-.377.245-.671.592-.845.999a2.18 2.18 0 0 0-.134 1.289c.086.433.303.83.622 1.144a2.3 2.3 0 0 0 1.169.612c.443.087.902.044 1.32-.125a2.272 2.272 0 0 0 1.026-.82c.251-.367.386-.799.386-1.24h7.953c.002.439.137.867.388 1.23.25.365.606.648 1.022.815.416.166.872.209 1.313.122.44-.086.844-.299 1.162-.61.317-.31.532-.706.62-1.136.086-.43.04-.877-.132-1.282a2.233 2.233 0 0 0-.837-.994 2.313 2.313 0 0 0-1.262-.374H4.996v-1.114H15.22Zm1.141-7.8-1.141 5.573H4.99V4.905H16.361Z"
              fill="currentColor"
              fill-rule="evenodd"
            />
          </svg>
        </span>
        Add To Cart
      </button>
      <button
        aria-label="Favorite product"
        class="flex items-center justify-center gap-2 rounded disabled:cursor-not-allowed disabled:opacity-50 bg-opacity-100 text-black group shadow-none p-4 bg-primary-50"
        disabled=""
        type="button"
      >
        <svg
          class="text-primary-100 transition-colors duration-200"
          fill="none"
          height="24"
          viewBox="0 0 84 80"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M54.285 9.72c2.857.003 5.652.816 8.05 2.341a14.652 14.652 0 0 1 5.467 6.258 14.344 14.344 0 0 1 1.136 8.171 14.466 14.466 0 0 1-3.56 7.469c-2.72 3.011-25.913 24.339-25.913 24.339S16.225 36.97 13.505 33.91a14.238 14.238 0 0 1-3.71-9.62 14.454 14.454 0 0 1 4.434-10.143 14.977 14.977 0 0 1 10.401-4.18c3.89 0 7.626 1.5 10.4 4.18a14.454 14.454 0 0 1 4.435 10.143c0-1.914.384-3.81 1.13-5.578a14.56 14.56 0 0 1 3.216-4.728 14.857 14.857 0 0 1 4.814-3.16 15.07 15.07 0 0 1 5.678-1.11l-.018.005ZM54.303 0a24.887 24.887 0 0 0-14.838 4.86A25.052 25.052 0 0 0 22.424.18C16.35.738 10.701 3.482 6.562 7.884 2.424 12.286.087 18.036.002 24.028-.082 30.02 2.09 35.83 6.103 40.344c2.778 3.13 21.072 19.954 26.622 25.037A9.988 9.988 0 0 0 39.481 68a9.987 9.987 0 0 0 6.755-2.62c5.52-5.077 23.738-21.858 26.551-24.983a24.106 24.106 0 0 0 5.932-12.45 23.902 23.902 0 0 0-1.897-13.62A24.418 24.418 0 0 0 67.704 3.9 25.05 25.05 0 0 0 54.285.004L54.303 0Z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
